### PR TITLE
New icons [breaking]

### PR DESCRIPTION
**Pull request by onishiweb**
Update to new square design and use new social icons.

Also removes Reddit sharing.

Before:
screen shot 2017-02-14 at 10 36 49

After:
screen shot 2017-02-15 at 15 27 21

After (hover):
screen shot 2017-02-14 at 10 37 48

### CODE REVIEW

File: src/scss/share.scss [hunk 1]
```diff
@@ -20,6 +20,7 @@
 		margin: 0 5px 0 0;
 		list-style-type: none;
 		line-height: $o-share-icon-size;
+		border: 1px solid #000000;
```
Comment (alicebartlett) [inline src/scss/share.scss]: #000000 👀

File: src/scss/share.scss [hunk 2]
```diff
@@ -98,22 +85,50 @@
...
+	$scheme: "ftsocial-v#{$o-share-social-icons-version}";
+
+	@if ($icon-name == 'mail' or $icon-name == 'link') {
+		$scheme: "fticon-v#{$o-share-ft-icons-version}";
+	}
+
+	$service-url: "#{$o-share-image-service-base-url}/#{$o-share-image-service-version}/images/raw/#{$scheme}:#{$icon-name}";
+	$query: "?source=o-share";
+
+	$hover-color: #ffffff;
```
Comment (alicebartlett) [inline src/scss/share.scss]: #ffffff 👀

**Comment by alicebartlett** [comment]
LGTM, A few colors that should probably be moved into color-use-cases. I'm really feeling this design from @olipowell too. :+1:

**Comment by onishiweb** [comment]
@alicebartlett @olipowell this has been updated with a few more changes since the first review and is now ready for a final review. Things that have changed:

- Fixed tests
- Lightened border colour (Oli requested)
- Remove the mail button from demo (Next use it for gift article so we can supply the styles but not the functionality)

File: src/scss/share.scss [hunk 3]
```diff
@@ -98,22 +87,54 @@
...
+	$hover-color: str-slice(ie-hex-str($hover-color), 4);
+	$tint: "&tint=%23#{$hover-color},%23#{$hover-color}";
+
+	// scss-lint:disable DuplicateProperty
+	.#{$classname}__action--#{$icon-name} i {
+		background-image: url($service-url + $query + "&format=svg");
+
+		// ie7/8 fallback. Uses the `\9` selector hack to target ie6-8.
+		// Hack is documented here: http://browserhacks.com/#hack-ab1bcc7b3a6544c99260f7608f8e845e
+		// It still needs to use the build service  <-- what does this comment mean?
```
Comment (alicebartlett) [inline src/scss/share.scss]: I think you've copied this `what does this comment mean` thing over from when i asked it of aberto in oicons...

Comment (alicebartlett) [inline src/scss/share.scss]: (for the record, i still don't know what it means...)

Comment (onishiweb) [inline src/scss/share.scss]: Erm yes I have and no I don't know what it means either. Shall I just remove that line?

File: src/scss/share.scss [hunk 4]
```diff
@@ -98,22 +87,54 @@
...
+		background-image: url($service-url + $query + "&format=png&width=#{$o-share-icon-size}")\9;
+
+		&:after {
```
Comment (alicebartlett) [inline src/scss/share.scss]: Can you add a comment about this being a hack to prevent a FOIC on hover as i don't think it's obvious

**Comment by alicebartlett** [comment]
A huge improvement!
I've left a few comments about comments. This also needs a migration guide to be added to the README.

It's really good to be able to close off those two open issues on o-share as well.

File: src/scss/share.scss (hunk 3 above)
Comment (alicebartlett) [inline src/scss/share.scss]: yep

**Comment by onishiweb** [comment]
I guess it will kind of close off the IG flavour icons issue, but it doesn't completely solve the touch target issue as these will only go up to 40x40 and not the required 44-48px 🔲  that is advised... Will talk to @olipowell some more about the solution

**Comment by onishiweb** [comment]
@alicebartlett readme has been updated to include the migration guide and fit to the new format

**Pull request merged**